test/js/parser.rb

```diff
@@ -4,7 +4,15 @@
 
 loop do
```

Comment:
This loop within a loop depends on the stdin terminating with "---\n", I'm not sure how/why but there's a case where "---\n" is never received and this inner loop never terminates.

Ideally we would find a why to fix the upstream problem of this never receiving an "---\n" rather than the solution provided here of breaking based on a counter.